packages/store/ts/config/v2/store.ts

```diff
@@ -26,20 +27,71 @@ export type resolveStoreTablesConfig<input, scope extends AbiTypeScope = AbiType
   readonly [key in keyof input]: resolveTableConfig<input[key], scope>;
 }>;
 
+export function resolveStoreTablesConfig<input, scope extends AbiTypeScope = AbiTypeScope>(
+  input: input,
+  scope: scope = AbiTypeScope as scope,
+): resolveStoreTablesConfig<input, scope> {
+  if (typeof input !== "object" || input == null) {
+    throw new Error(`Expected tables config, received ${JSON.stringify(input)}`);
+  }
+
+  return Object.fromEntries(
+    Object.entries(input).map(([key, table]) => [key, resolveTableConfig(table, scope)]),
+  ) as resolveStoreTablesConfig<input, scope>;
+}
+
 export type scopeWithUserTypes<userTypes, scope extends AbiTypeScope = AbiTypeScope> = UserTypes extends userTypes
   ? scope
   : userTypes extends UserTypes
     ? extendScope<scope, userTypes>
     : scope;
 
+function isUserTypes(userTypes: unknown): userTypes is UserTypes {
+  return (
+    typeof userTypes === "object" &&
+    userTypes != null &&
+    Object.values(userTypes).every((type) => isSchemaAbiType(type))
+  );
+}
+
+export function scopeWithUserTypes<userTypes, scope extends AbiTypeScope = AbiTypeScope>(
+  userTypes: userTypes,
+  scope: scope = AbiTypeScope as scope,
+): scopeWithUserTypes<userTypes, scope> {
+  return (isUserTypes(userTypes) ? extendScope(scope, userTypes) : scope) as scopeWithUserTypes<userTypes, scope>;
+}
+
+function isEnums(enums: unknown): enums is Enums {
+  return (
+    typeof enums === "object" &&
+    enums != null &&
+    Object.values(enums).every((item) => Array.isArray(item) && item.every((element) => typeof element === "string"))
+  );
+}
+
 export type scopeWithEnums<enums, scope extends AbiTypeScope = AbiTypeScope> = Enums extends enums
   ? scope
   : enums extends Enums
     ? extendScope<scope, { [key in keyof enums]: "uint8" }>
     : scope;
 
+export function scopeWithEnums<enums, scope extends AbiTypeScope = AbiTypeScope>(
+  enums: enums,
+  scope: scope = AbiTypeScope as scope,
+): scopeWithEnums<enums, scope> {
+  if (isEnums(enums)) {
```

Comment:
I reason I was using mostly things like `validateEnums` where it returns `asserts enums is Enums` was so we could both type narrow and throw runtime errors for e.g. invalid enums

not blocking but any reason this approach changed?

Reply:
in this case i wanted to mirror the type behavior which extends the scope with enums if enums are provided and just returns the old scope if none are provided, instead of throwing an error